cmd/sample-metrics-plugin/internal/plugin/plugin.go

```diff
@@ -0,0 +1,185 @@
+package plugin
```

Comment:
I think it is better to centralize test/e2e related resources in the `/test` folder. The reason is that usually we consider the `cmd` folder to contain the official released binaries for the application